### PR TITLE
Task Board component upgrade

### PR DESCRIPTION
**Pull request by rana-nouman**

### CODE REVIEW

File: app/components/app-ui/task-board.hbs [hunk 1]
```diff
@@ -0,0 +1,106 @@
+<div class="board-container" {{initialize-sortable group=@group scroll=@scroll scrollSensitivity=@scrollSensitivity scrollSpeed=@scrollSpeed sort=@sort forceFallback=@forceFallback disableSortable=@disableSortable animationSpeed=@animationSpeed dragClass=@dragClass chosenClass=@chosenClass}}>
+{{#each @milestones as |milestone|}}
+    {{#if (or (eq milestone.status 'in_progress') (eq milestone.status 'planned'))}}
+      <div class="box">
+        <div class="box-header with-border">
+          <strong>{{t (concat "views.app.milestone.lists.type." milestone.milestoneType)}} {{milestone.name}}</strong>
+          <div class="box-tools pull-right filters">
+            {{text-field
```
Comment (gollomer) [inline app/components/app-ui/task-board.hbs:10]: convert to glimmer component

File: app/components/app-ui/task-board.hbs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+                  <div class="lane box-body" data-field-lane-group={{@group}}>
+                    {{#each milestone.issues as |issue|}}
+                      {{#if (eq issue.status status)}}
+                        <div class="item" style="{{laneitem-color issue.issuetype.name}}" data-field-issue-id="{{issue.id}}" data-field-issue-status="{{issue.status}}" data-field-issue-milestone="{{issue.milestoneId}}">
+                          <h4>{{priority-icon priority=(readonly issue.priority)}} <a href="#" {{action "openIssue"issue}}>#{{issue.issueNumber}} - {{issue.subject}}</a></h4>
+                          <p class="description">{{safe-html issue.description}}</p>
+                          <p>
+                            <span class="badge {{issue.status}}">{{t (concat "views.app.issue.lists.status." issue.status)}}</span>
+                            <span><img class="img-circle" src="http://projects4me/userimage/get/{{issue.assignedTo.id}}" alt="{{issue.assignedTo.name}}" title="{{issue.assignedTo.name}}"></span>
```
Comment (gollomer) [inline app/components/app-ui/task-board.hbs:43]: Change the image path

File: app/components/app-ui/task-board.hbs [hunk 3]
```diff
@@ -0,0 +1,106 @@
+<div class="board-container" {{initialize-sortable group=@group scroll=@scroll scrollSensitivity=@scrollSensitivity scrollSpeed=@scrollSpeed sort=@sort forceFallback=@forceFallback disableSortable=@disableSortable animationSpeed=@animationSpeed dragClass=@dragClass chosenClass=@chosenClass}}>
```
Comment (gollomer) [inline app/components/app-ui/task-board.hbs:1]: Improve endline consistency in the file

File: app/modifiers/initialize-sortable.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+     * @method get
+     * @return String
+     * @public
+     */
+    get chosenClass() {
+        return this.args.named.chosenClass;
+    }
+
+    /**
+     * This property is used to keep in track that whether element is dragging or not.
```
Comment (gollomer) [inline app/modifiers/initialize-sortable.js]: Check if this is still being used to not. Install ESLint plugin in VSCode.

File: app/modifiers/initialize-sortable.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+                scroll: _self.scroll,
+                scrollSensitivity: _self.scrollSensitivity,
+                scrollSpeed: _self.scrollSpeed,
+                sort: _self.sort,
+                forceFallback: _self.forceFallback,
+                disableSortable: _self.disableSortable,
+                animation: _self.animationSpeed,
+                dragClass: _self.dragClass,
+                chosenClass: _self.chosenClass,
+                onStart: (evt) => {
```
Comment (gollomer) [inline app/modifiers/initialize-sortable.js:174]: isolate these function as _onStart, _onEnd, etc. functions.

File: app/modifiers/initialize-sortable.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+    }
+
+    /**
+     * This function is called when an item of one lane is shifted to another lane
+     * and is used to set the height of parent element by getting max size of
+     * lane and set it to its parent element.
+     * @method setParentHeight
+     */
+    setParentHeight() {
+        let parentElArray = document.querySelectorAll('div.board.row > div > div.box > div.box-body');
```
Comment (gollomer) [inline app/modifiers/initialize-sortable.js]: identify using a direct classname or identifier.

File: app/modifiers/initialize-sortable.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+        parentElArray.forEach((parentElement) => {
+            let lanes = [...parentElement.children];
+            let heightArray = [];
+            lanes.forEach((lane) => {
+                let laneBody = lane.querySelector('div.lane.box-body');
+                let items = [...laneBody.children];
+                let sum = 0;
+                items.forEach((item) => {
+                    let padding = 3.2;
```
Comment (gollomer) [inline app/modifiers/initialize-sortable.js]: calculate this value.

File: app/modifiers/initialize-sortable.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+                    sum += item.getBoundingClientRect().height + padding;
+                })
+                heightArray.push(sum);
+            });
+            let max = (Math.max(...heightArray) + 85);
```
Comment (gollomer) [inline app/modifiers/initialize-sortable.js]: calculate

File: app/components/app-ui/task-board.hbs [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+{{#each @milestones as |milestone|}}
+    {{#if (or (eq milestone.status 'in_progress') (eq milestone.status 'planned'))}}
+      <div class="box">
```
Comment (gollomer) [inline app/components/app-ui/task-board.hbs]: lets add the milestone identifier

File: app/styles/modules/board.scss [hunk 10]
```diff
@@ -8,128 +8,165 @@
...
+        height: 100%;
+
+        .box {
+          height: 100%;
+          box-shadow: none;
+          
+          .box-body {
+            min-height: 45rem;
+            padding: 0rem;
+            border: 2px solid #fff;
```
Comment (gollomer) [inline app/styles/modules/board.scss]: lets use rem

File: app/styles/modules/board.scss [hunk 11]
```diff
@@ -8,128 +8,165 @@
...
+
+              .img-circle {
+                width: 4rem;
+                right: 0;
+                position: absolute;
+                margin: -1.5rem 1rem;
+              }
+
+              .fa-ban {
+                color: #d33724;
```
Comment (gollomer) [inline app/styles/modules/board.scss:57]: Lets handle it with mixins

File: app/styles/modules/board.scss [hunk 12]
```diff
@@ -8,128 +8,165 @@
...
+
+            .curr-item {
+              border-top: 1px dashed $primaryColor;
+              border-bottom: 1px dashed $primaryColor;
+              border-right: 1px dashed $primaryColor;
+            }
+          }
+
+          .box-body-border {
+            border: 2px dashed #000;
```
Comment (gollomer) [inline app/styles/modules/board.scss]: rem + variables

File: app/modifiers/initialize-sortable.js [hunk 13]
```diff
@@ -0,0 +1,269 @@
+import Modifier from 'ember-modifier';
```
Comment (gollomer) [inline app/modifiers/initialize-sortable.js:1]: Lets write its test as well.

**Review by gollomer — CHANGES_REQUESTED** [review]

File: app/components/app-ui/task-board.hbs (hunk 3 above)
Comment (nouman-rt) [reply app/components/app-ui/task-board.hbs:1]: http://ember.projects4me/app/project/add12ff55acb-1902-6b9c-30d5411f7729/issue/3186

File: app/components/app-ui/task-board.hbs (hunk 1 above)
Comment (rana-nouman) [reply app/components/app-ui/task-board.hbs:10]: http://dev.projects4.me/app/project/add12ff55acb-1902-6b9c-30d5411f7729/issue/3185

File: app/components/app-ui/task-board.hbs (hunk 2 above)
Comment (rana-nouman) [reply app/components/app-ui/task-board.hbs:43]: http://dev.projects4.me/app/project/add12ff55acb-1902-6b9c-30d5411f7729/issue/3186

File: app/styles/modules/board.scss (hunk 11 above)
Comment (rana-nouman) [reply app/styles/modules/board.scss:57]: http://dev.projects4.me/app/project/add12ff55acb-1902-6b9c-30d5411f7729/issue/3187

**Review by gollomer — APPROVED** [review]

**Pull request merged**